openfl/federated/task/runner_keras.py

```diff
@@ -182,7 +182,16 @@ def train_(self, batch_generator, metrics: list = None, **kwargs):
         #  initialization (build_model).
         #  If metrics are added (i.e. not a subset of what was originally
         #  defined) then the model must be recompiled.
-        results = self.model.get_metrics_result()
+        try:
+            results = self.model.get_metrics_result()
+        except ValueError:
+            if "batch_size" in kwargs:
+                batch_size = kwargs["batch_size"]
+            else:
+                batch_size = 1
+            # evaluation needed before metrics can be resolved
+            self.model.evaluate(self.data_loader.get_valid_loader(batch_size), verbose=1)
```

Comment:
For other reviewers reference, this is needed for the scenario when the collaborator dies between the `aggregated_model_validation` and `train` task. If the collaborator restarts it will immediately be assigned `train`, and will not be able to resolve the necessary metrics. The names of needed metrics can be resolved by running the `evaluate` function on the model, but these metric values will not be sent to the aggregator.